packager/media/formats/mp4/segmenter.cc

```diff
@@ -165,43 +165,7 @@ Status Segmenter::FinalizeSegment(size_t stream_id,
       return Status::OK;
   }
 
-  MediaData mdat;
-  // Data offset relative to 'moof': moof size + mdat header size.
-  // The code will also update box sizes for moof_ and its child boxes.
-  uint64_t data_offset = moof_->ComputeSize() + mdat.HeaderSize();
-  // 'traf' should follow 'mfhd' moof header box.
-  uint64_t next_traf_position = moof_->HeaderSize() + moof_->header.box_size();
-  for (size_t i = 0; i < moof_->tracks.size(); ++i) {
-    TrackFragment& traf = moof_->tracks[i];
-    if (traf.auxiliary_offset.offsets.size() > 0) {
-      DCHECK_EQ(traf.auxiliary_offset.offsets.size(), 1u);
-      DCHECK(!traf.sample_encryption.sample_encryption_entries.empty());
-
-      next_traf_position += traf.box_size();
-      // SampleEncryption 'senc' box should be the last box in 'traf'.
-      // |auxiliary_offset| should point to the data of SampleEncryption.
-      traf.auxiliary_offset.offsets[0] =
-          next_traf_position - traf.sample_encryption.box_size() +
-          traf.sample_encryption.HeaderSize() +
-          sizeof(uint32_t);  // for sample count field in 'senc'
-    }
-    traf.runs[0].data_offset = data_offset + mdat.data_size;
-    mdat.data_size += static_cast<uint32_t>(fragmenters_[i]->data()->Size());
-  }
-
-  // Generate segment reference.
-  sidx_->references.resize(sidx_->references.size() + 1);
-  fragmenters_[GetReferenceStreamId()]->GenerateSegmentReference(
-      &sidx_->references[sidx_->references.size() - 1]);
-  sidx_->references[sidx_->references.size() - 1].referenced_size =
-      data_offset + mdat.data_size;
-
   const uint64_t moof_start_offset = fragment_buffer_->Size();
```

Comment:
Is this moof_start_offset still correct after your change? Now that writing of moof atom is moved inside fragment, wouldn't the offset also change?

Reply:
The math for the `moof_start_offset` is still correct because Packager assumes that there is only 1 fragment per segment, so the key frame will always be at the start of the segment. With this assumption, the `fragment_buffer_->Size()` is always 0 at this point in the code. This PR is meant to be a stepping stone and is purposed to move the header data to the fragment level to tee us up for the next part. The next diff, which will actually implement the multiple fragments per segment feature, will have to address the `moof_start_offset`.

Reply:
Makes sense, thanks for clarifying. I usually add comment in code for things like this (for both your reference and reviewers), but that's just my style :-)

Reply:
I agree.  Please add a comment!  You never know when something like that will be forgotten.  Make it easy for others and for your future self to pick up the work after an unforeseen interruption.

Reply:
Great point! Will do now :)